hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -163,6 +167,22 @@ private Executor newExecutor() {
                 ExecutorType.CONCRETE);
     }
 
+    private Executor newClientQueryExecutor() {
```

Comment:
Why is a new threadpool introduced? Can't it be processed by the generic threadpool?